### ISSUE TITLE
RMG explores a PDep network and adds it to the core although a similar library reaction exists

### ISSUE THREAD

**Issue by alongd** [ReactionMechanismGenerator/RMG-Py#1267]
Consider the following reactions in a chemkin file generated by RMG:
```
! Reaction index: Chemkin #195; RMG #515
! Library reaction: Sulfur/GlarborgH2S
! Flux pairs: HSSO2(53), SH(39); HSSO2(53), SO2(42); 
HSSO2(53)+M=SH(39)+SO2(42)+M                        1.000e+17 0.000     3.000    

! Reaction index: Chemkin #196; RMG #6348
! PDep reaction: PDepNetwork #421
! Flux pairs: HSSO2(53), SH(39); HSSO2(53), SO2(42); 
HSSO2(53)(+M)=SH(39)+SO2(42)(+M)                    1.000e+00 0.000     0.000    
    TCHEB/ 350.000   1000.000 /
    PCHEB/ 19.738    49.346   /
    CHEB/ 6 4/
    CHEB/ -1.100e+00   1.606e-01    -3.268e-03   -1.300e-04  /
    CHEB/ 6.899e+00    2.653e-02    1.824e-03    1.968e-05   /
    CHEB/ -1.146e-01   -8.191e-04   1.237e-04    2.396e-05   /
    CHEB/ -1.231e-02   -9.293e-04   -7.876e-05   -1.074e-06  /
    CHEB/ 2.249e-03    -2.136e-04   -2.770e-05   -2.130e-06  /
    CHEB/ 1.428e-03    -6.186e-07   -2.419e-06   -4.937e-07  /
```

This often happens when a library reaction is defined at the low pressure limit (`kinetics = ThirdBody`). From RMG's current point of view, these aren't duplicate reactions since one has `+M`, while the other, explored by RMG, has `(+M)`. However, this common situation doesn't pass the Chemkin per-processor  (`Error...undeclared duplicate to reaction number   195`).

Theoretically, these could be declared as DUPLICATE if they pass through different TSs. However, since we commonly don't know the TS of the library reaction, RMG should assume they are the same and not re-generate it. The problem I see here is that we would expect Cantherm to use the library rate for the network exploration, however the library does not supply a high-P limit in the `ThirdBody` kinetics format. So Cantherm has no choice but regenerating it.

We should decide what to do in situations like this. One proposal is to allow Cantherm to re-generate the reaction using families (as it anyway does now) since it might need it for the entire network, however network reactions should be recognized as existing even if the `+M`/`(+M)` markings are different. Other reactions from the network can be added as appropriate, but the recognized existing reaction should then be dropped. This, however, might lead to slight kinetic disagreements since different kinetics are used for the same reaction in a model (the library and the underlying network reaction).

This was generated on the chargedAT branch. I'll try to upload an input file that reproduces this on the master branch.

**Comment by mliu49** [comment]
The fact that this doesn't pass the Chemkin pre-processor is due RMG failing to properly mark duplicates, which is done [here](https://github.com/ReactionMechanismGenerator/RMG-Py/blob/master/rmgpy/chemkin.py#L1733). We assume that reactions of differing classes (e.g. library vs. family) don't need to be checked, which is clearly incorrect.

I agree with your other points, though I don't know what the best solution would be either.

**Comment by alongd** [comment]
solved by #1296
